src/app/article/[articleId]/layout.tsx

```diff
@@ -1,7 +1,36 @@
+import { Metadata } from "next";
+
 import React, { ReactNode } from "react";
 
-import TopBar from "@common/components/TopBar";
+import { ApiResponse } from "@api/api-config";
+
 import ArticleBottomButton from "@article/components/ArticleBottomButton/indext";
+import { API_ROUTE } from "@article/remotes/api";
+import { ArticleDetail } from "@article/types";
+
+import TopBar from "@common/components/TopBar";
+
+export async function generateMetadata({
+  params,
+}: {
+  params: { articleId: string };
+}): Promise<Metadata> {
+  const articleId = params.articleId;
+  const response = await fetch(
+    `${process.env.NEXT_PUBLIC_API_BASE_URL}${API_ROUTE.ARTICLE(articleId)}`,
+  );
+  const { data: articleInfo }: ApiResponse<ArticleDetail> = await response.json();
+
+  return {
+    title: articleInfo.title,
+    openGraph: {
+      title: articleInfo.title,
+    },
+    twitter: {
+      title: articleInfo.title,
+    },
+  };
```

Comment:
```suggestion
 const { data: articleInfo }: ApiResponse<ArticleDetail> = await response.json();
const { title} = articleInfo;
  return {
    title,
    openGraph: {
      title,
    },
    twitter: {
      title,
    },
  };
```
위에서도 한번더 구조분해 할당하면 어떨까요?!